image_segmentation/pytorch/runtime/training.py

```diff
@@ -0,0 +1,132 @@
+from tqdm import tqdm
+
+import torch
+from torch.optim import Adam, SGD
+from torch.cuda.amp import autocast, GradScaler
+
+from runtime.distributed_utils import get_rank, reduce_tensor, get_world_size
+from runtime.inference import evaluate
+from runtime.logging import mllog_event, mllog_start, mllog_end, CONSTANTS
+
+
+def get_optimizer(params, flags):
+    if flags.optimizer == "adam":
+        optim = Adam(params, lr=flags.learning_rate, weight_decay=flags.weight_decay)
+    elif flags.optimizer == "sgd":
+        optim = SGD(params, lr=flags.learning_rate, momentum=flags.momentum, nesterov=True,
+                    weight_decay=flags.weight_decay)
+    elif flags.optimizer == "lamb":
+        import apex
```

Comment:
I'm not 100% certain whether apex (along with amp) will need CUDA. Do you know if it runs on CPU only?

Reply:
Good point, I'll verify

Reply:
It seems it doesn't work. 
I'll have a look whether there is a non-cuda implementation of lamb, but fyi: I haven't found a better configuration with lamb that sgd

Reply:
Is LAMB needed then?

Reply:
Just because I haven't found a config it doesn't mean that there isn't any :) 
I'd leave it in the code, maybe it will be useful in the future.